query/src/main/scala/filodb/query/exec/MetadataRemoteExec.scala

```diff
@@ -88,6 +89,34 @@ case class MetadataRemoteExec(queryEndpoint: String,
     val rv = IteratorBackedRangeVector(CustomRangeVectorKey.empty, NoCloseCursor(rows.iterator), None)
     // dont add this size to queryStats since it was already added by callee use dummy QueryStats()
     val srv = SerializedRangeVector(rv, builder, RECORD_SCHEMA, queryWithPlanName(queryContext), dummyQueryStats)
+
+    // NOTE: We are using the RESULT_SCHEMA definitions to determine the iteration of shardKeyPrefix in v1 result.
+    // Hence, we are sending the older result schema which was used for V1 Cardinality API
+    QueryResult(id, RESULT_SCHEMA_V1, Seq(srv))
+  }
+
+  /**
+  * @param response Metadata Response from the remote query server API call.
+  * @param id QueryId
+  * @return We convert the TsCardinalitiesSamplV2 response to QueryResult which can be appropriately parsed
+   *         by the query service and return the response to the user
+  */
+  private def mapTsCardinalitiesResponseV2(response: MetadataSuccessResponse, id: String): QueryResponse = {
+    import NoCloseCursor._
+    import TsCardinalities._
+    import TsCardExec._
+
+    val RECORD_SCHEMA = SerializedRangeVector.toSchema(RESULT_SCHEMA.columns)
+
+    val rows = response.data.asInstanceOf[Seq[TsCardinalitiesSamplV2]]
+      .map { ts =>
+        val prefix = SHARD_KEY_LABELS.take(ts.group.size).map(l => ts.group(l))
+        val counts = CardCounts(ts.cardinality("active"), ts.cardinality("shortTerm"), ts.cardinality("longTerm"))
```

Comment:
Does this need to be consistent with this?:
```
  val RESULT_SCHEMA = ResultSchema(Seq(ColumnInfo("group", ColumnType.StringColumn),
                                        ColumnInfo("active", ColumnType.IntColumn),
                                        ColumnInfo("total", ColumnType.IntColumn),
                                        ColumnInfo("longterm", ColumnType.IntColumn)), 1)
```

Reply:
Great catch. Thanks Alex :) Updating it